database/gormdb_course.go

```diff
@@ -84,6 +84,15 @@ func (db *GormDB) GetCourseByOrganizationID(did uint64) (*pb.Course, error) {
 	return &course, nil
 }
 
+// GetCourseBycodeAndYear returns course with given code and year
+func (db *GormDB) GetCourseByCodeAndYear(courseCode string, courseYear uint32) (*pb.Course, error) {
```

Comment:
To make this more flexible, we should take a `*pb.Course`, letting the caller fill the fields that should be queried.
Maybe it should be renamed to `GetCourseByCourseQuery` or something better. It looks a bit difficult to convert `GetCourse` into a method that works seamlessly, although that would be better IMO...

Reply:
This all said. Maybe this should be implemented as a Join operation on the db, where we query both course and user in the same transaction. This way we would avoid two database methods... (but maybe not worth it, since it increases complexity a bit...)

Reply:
Alternatively, we could remove this method and keep the one fetching the user. We can pass the course query to the `GetUserByLogin` method (while renaming it to something like `GetUserByCourseAndLogin`). This way we can return an error both when there is no such course, no such user, or user is not enrolled, directly from here, without extra database calls and checks. If every condition is met, we would return the user already with the preloaded course. This way we also end up with just one method.

Reply:
Sounds good to me!  Maybe we could just call it `GetUserByCourse(login string, course *pb.Course)` implying that it will also require the `login`, but that it's the course that counts... but it is debatable what to call it...